Addresses all Code Analyzer CA1801 issues

dominoFire opened the pull request:
<!-- DO NOT MODIFY OR DELETE THIS TEMPLATE. IT IS USED IN AUTOMATION. -->
## Bug

<!-- Search https://github.com/NuGet/Home/issues, and create one if you can't find a suitable issue. -->
<!-- Paste the full link, like https://github.com/nuget/home/issues/1000. GitHub will render is neatly. -->
Fixes: https://github.com/NuGet/Client.Engineering/issues/640

Regression? Last working version: No

## Description
<!-- Add details about the fix. Include any information that would help the maintainer review this change effective. -->

All CA1801 warnings are addressed using the following method:

```
For each Code Analyzer CA1801 warning, do the following:
1) if it is shipped Public API:
  1.a) Keep Code Analyzer warning, write justification and provide a xmldoc saying that parameter is not used.
2) otherwise:
  2.a) If it is a CancellationToken:
    2.a.i) either call ThrowIfCancellationRequested() or use as an argument inside the method body.
  2.b) otherwise:
    2.b.i) Verify no usage in commit history or multitargeting scenarios. If nothing is found, remove the parameter.
```

The following [spreadsheet](https://microsoft-my.sharepoint.com/:x:/p/feaguila/EQNs0t9FQohOoKtSvGlAZYoBD1-pw4FXAA2Dn7wLDgd2pw?e=AKjB8D) contains the actions taken for each warning listed in the issue.



## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  <!-- Describe why you haven't added automation. -->
  - [x] Test exception: Since there's no product change, no new tests are added. Let me know if I need to add test where cancellation tokens cases.
  - **OR**
  - [ ] N/A <!-- Infrastructure, documentation etc. -->

- **Documentation**
  <!-- Please link the PR/issue if appropriate -->
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

Commit:
First batch of CA1801 fixes

Commit:
Addressed NuGet.Procol CA1801 unused params

Commit:
Addressed NuGet.DependencyResolver.Core

Commit:
Addressed NuGet.Commands CA1801 issues

Commit:
Used all cancellationTokens in parameters

Commit:
Addressted some fixes in N.PM.VS and N.PM

Commit:
Adding suppression and fixing compile errors

Commit:
Addressed all CA1801

Commit:
Fixed format

Commit:
Filled justifications for remaining CA1801 issues

Commit:
Making NuGet.Packaging.PackageExtractionResult.

Commit:
Added notes to NuGet.Packaging.PackageExtractionResult

Commit:
Removed param in call

erdembayar commented on src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectSystems/VsMSBuildProjectSystem.cs:
```diff
@@ -708,7 +708,7 @@ private IEnumerable<string> GetChildItems(string path, string filter, string des
                 await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
-                var childItems = await EnvDTEProjectUtility.GetChildItems(VsProjectAdapter.Project, path, filter, VsProjectTypes.VsProjectItemKindPhysicalFile);
+                var childItems = await EnvDTEProjectUtility.GetChildItems(VsProjectAdapter.Project, path, filter, desiredKind);
```
Now from line#702 it passes `VsProjectTypes.VsProjectItemKindPhysicalFolder` for `desiredKind`. It's different from previous `VsProjectTypes.VsProjectItemKindPhysicalFile`.
Did you make sure it still works same as before?

dominoFire replied:
In this [commit](https://github.com/NuGet/NuGet.Client/commit/63b3c65986f5434b2dc44d32057a887ba967ccda), `GetFiles()` and `GetDirectories()` were added to this class.

Later, in another [commit](https://github.com/NuGet/NuGet.Client/commit/7985322db49b11340ad4bf0c73247c9bd0096621), `GetChildItems()` is introduced as a refactor of `GetFiles()` and `GetDirectories()`.

Looking at the first two functions, it seems that the correct parameter is `desiredKind`.

erdembayar commented on src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPathContextProvider.cs:
```diff
@@ -279,6 +279,8 @@ internal IVsPathContext GetSolutionPathContext()
         private async Task<IVsPathContext> GetPathContextFromAssetsFileAsync(
             BuildIntegratedNuGetProject buildIntegratedProject, CancellationToken token)
         {
+            token.ThrowIfCancellationRequested();
```
I see many `token.ThrowIfCancellationRequested();` are introduced. I guess we're going to have more `cancellation token exception` than before. Do we need to make other changes to reduce spamming of telemetry? Maybe connected to https://github.com/NuGet/Client.Engineering/issues/776

dominoFire replied:
It might be. But, that's a better way to diagnose possible problems. We just need to monitor telemetry.

erdembayar commented on src/NuGet.Core/NuGet.Packaging/PackageArchiveReader.cs:
```diff
@@ -78,8 +78,8 @@ public PackageArchiveReader(Stream stream)
         /// <param name="stream">Nupkg data stream.</param>
-        /// <param name="frameworkProvider">Framework mapping provider for NuGetFramework parsing.</param>
-        /// <param name="compatibilityProvider">Framework compatibility provider.</param>
+        /// <param name="frameworkProvider">Framework mapping provider for NuGetFramework parsing. No used in this constructor</param>
```
`Not used in this constructor`.

dominoFire replied:
I will fix in next commit.

erdembayar commented on src/NuGet.Core/NuGet.Packaging/PackageArchiveReader.cs:
```diff
@@ -78,8 +78,8 @@ public PackageArchiveReader(Stream stream)
-        /// <param name="frameworkProvider">Framework mapping provider for NuGetFramework parsing.</param>
-        /// <param name="compatibilityProvider">Framework compatibility provider.</param>
+        /// <param name="frameworkProvider">Framework mapping provider for NuGetFramework parsing. No used in this constructor</param>
+        /// <param name="compatibilityProvider">Framework compatibility provider. No used in this constructor</param>
```
`Not used in this constructor`.

dominoFire replied:
I will fix it in next commit.

erdembayar commented on src/NuGet.Core/NuGet.ProjectModel/PackageSpecReferenceDependencyProvider.cs:
```diff
@@ -149,13 +149,13 @@ public Library GetLibrary(LibraryRange libraryRange, NuGetFramework targetFramew
             {
                 // Additional library properties
-                AddLibraryProperties(library, packageSpec, targetFramework, msbuildPath);
+                AddLibraryProperties(library, packageSpec, targetFramework); // use library if you need msbuildpath
```
Do we need this new comment?

dominoFire replied:
Removed.

erdembayar commented on src/NuGet.Core/NuGet.Packaging/GlobalSuppressions.cs:
```diff
@@ -185,12 +180,12 @@
 [assembly: SuppressMessage("Build", "CA1031:Modify 'Verify' to catch a more specific allowed exception type, or rethrow the exception.", Justification = "<Pending>", Scope = "member", Target = "~M:NuGet.Packaging.Signing.Signature.Verify(NuGet.Packaging.Signing.Timestamp,NuGet.Packaging.Signing.SignatureVerifySettings,NuGet.Common.HashAlgorithmName,System.Security.Cryptography.X509Certificates.X5 ...
 [assembly: SuppressMessage("Build", "CA1031:Modify 'IsUntrustedRootAllowed' to catch a more specific allowed exception type, or rethrow the exception.", Justification = "<Pending>", Scope = "member", Target = "~M:NuGet.Packaging.Signing.SignatureTrustAndValidityVerificationProvider.IsUntrustedRootAllowed(NuGet.Packaging.Signing.Signature)~System.Boolean")]
-[assembly: SuppressMessage("Build", "CA1801:Parameter signatureCentralDirectoryHeader of method AssertSignatureEntryCommonHeaderFields is never used. Remove the parameter or use it in the method body.", Justification = "<Pending>", Scope = "member", Target = "~M:NuGet.Packaging.Signing.SignedPackageArchiveIOUtility.AssertSignatureEntryCommonHeaderFields(System.IO.BinaryReader,NuGet.Packaging.Sign ...
+[assembly: SuppressMessage("Build", "CA1825:Avoid unnecessary zero-length array allocations.  Use Array.Empty<byte>() instead.", Justification = "<Pending>", Scope = "member", Target = "~M:NuGet.Packaging.Signing.SignedPackageArchiveUtility.GetPackageContentHash(System.IO.BinaryReader)~System.String")]
```
Just wondering why new one pop-up here?

dominoFire replied:
Without those new supression, I was unable to compile the project.

dominoFire replied:
Will remove in next iteration.

erdembayar reviewed (commented):
Just few comments.

Commit:
Feedback 1

zivkan commented on src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreOperationLogger.cs:
```diff
@@ -487,6 +487,8 @@ private class WaitDialogProgress : RestoreOperationProgressUI
                 JoinableTaskFactory jtf,
                 CancellationToken token)
             {
+                token.ThrowIfCancellationRequested();
+
                 await jtf.SwitchToMainThreadAsync();
```
Throwing in case of cancellation requested I think would be more beneficial after switching to the main thread, or maybe checking before and after.

Imagine when StartAsync is called, the token is not cancelled, but it takes 1+ seconds to switch to the UI thread, during which time the token is cancelled. We wouldn't want to do "expensive" work on the UI thread in this case.

dominoFire replied:
We can also pass the token to `SwitchToMainThreadAsync(token)` .

https://github.com/microsoft/vs-threading/blob/main/doc/cookbook_vs.md#canceling-requests-for-the-main-thread

Then, we could have the following code pattern:

```c#
// ... some code 
token.ThrowIfCancellationRequested();

await jtf.SwitchToMainThreadAsync(token);

token.ThrowIfCancellationRequested();
// ... more code 
```

What do you think ?

dominoFire replied:
Added more calls.

zivkan replied:
Passing the token to `SwitchToMainThreadAsync` will ~stop the `await` waiting for the result, but it won't prevent the continuation running on the main thead and using time~ make the continuation run on a background thread, but won't stop the rest of the method from running. So, that `ThrowIfCancellationRequested` after the switch is still useful.

Commit:
using token before, during and after SwitchToMainThreadAsync(token) call

nkolev92 commented on src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs:
```diff
@@ -558,7 +558,7 @@ public static MsBuildToolset GetMsBuildToolset(string userVersion, IConsole cons
         /// </summary>
         /// <param name="userVersion">version string as passed by user (so may be empty)</param>
-        /// <param name="console">console for status reporting</param>
+        /// <param name="console">console for status reporting. Not used</param>
```
Why don't we just remove it?

dominoFire replied:
Removed

nkolev92 commented on src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreOperationLogger.cs:
```diff
@@ -554,7 +558,11 @@ private class StatusBarProgress : RestoreOperationProgressUI
                 CancellationToken token)
             {
-                await jtf.SwitchToMainThreadAsync();
+                token.ThrowIfCancellationRequested();
```
This seems really excessive to me. 
https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.threading.joinabletaskfactory.switchtomainthreadasync?view=visualstudiosdk-2019 says that the check will happen in the switch to main thread.

dominoFire replied:
Parameters section says:

> A token whose cancellation will immediately schedule the continuation on a threadpool thread (if the transition to the main thread is not already complete). The token is ignored if the caller was already on the main thread.

... but Exception section says:

> Thrown back at the awaiting caller if cancellationToken is canceled, even if the caller is already on the main thread.

So, is the token ignored if it's on the main thread?

nkolev92 replied:
Check it before and pass it to the switch to main thread async?

I get the theoretical benefit, but at this point, it just adds clutter imo.

dominoFire replied:
Removed extra cancelation token call.

nkolev92 commented on src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs:
```diff
@@ -32,6 +32,15 @@ public class LockFileBuilder
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="previousLockFile"></param>
```
Missing documentation here?

dominoFire replied:
Fixed

nkolev92 commented on src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreTargetGraph.cs:
```diff
@@ -90,6 +90,16 @@ public static RestoreTargetGraph Create(IEnumerable<GraphNode<RemoteResolveResul
+        /// <summary>
+        /// Factory method
+        /// </summary>
+        /// <param name="runtimeGraph"></param>
```
I don't think there's much value in having the name without a description.

dominoFire replied:
Added docs.

nkolev92 commented on src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/BuildAssetsUtils.cs:
```diff
@@ -421,6 +421,18 @@ public static string GetMSBuildFilePathForPackageReferenceStyleProject(PackageSp
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="project"></param>
```
Not sure what the value is of a description that doesn't really add anything.

nkolev92 replied:
I see lots of 👎 by @donnie-msft here. 

`<summary></summary>` is the same as having nothing to the intellisense documentation. 

I don't see the point of adding an empty summary element.

donnie-msft replied:
That's correct, but the template generates all the parameters. Comments are not compiled. Our codebase is full of empty parameters. If you have a proposal for a Coding Guideline, we should discuss that separately from this PR. This PR is following existing conventions.

nkolev92 replied:
There's a lot of code checked in that doesn't follow the current coding conventions.
There's also a lot of coding conventions that are not called out into the coding guidelines. 

As is, currently this resolves and arguably low importance warning suppression at the cost of adding lines that provide any value, just taking space on screen. Yes I know you can collapse comments, but we have a combination of good and bad comments and collapsing them defeats the benefits of having them.

If you look at some older PRs you will notice some of these code comment standards being applied by reviewers. 

I have created https://github.com/NuGet/NuGet.Client/pull/4051 to codify this.

dominoFire replied:
Added some more docs.

nkolev92 commented on src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/NoOpRestoreUtilities.cs:
```diff
@@ -44,6 +44,12 @@ private static string GetBuildIntegratedProjectCacheFilePath(RestoreRequest requ
+        /// </summary>
+        /// <param name="cacheRoot">Path to cache root</param>
+        /// <param name="projectPath">Not used</param>
+        /// <returns></returns>
```
missing description.

donnie-msft replied:
When it's obvious, like "Get file path", you can just assume it's the file path in the string. I'm fine with leaving those blank.
If anything, explain what `null` means or `empty` means, etc.

nkolev92 replied:
The description for empty returns is ignored by the intellisense.

Whether you put `<returns></returns>` or ``, it's the same effect to expanded documentation.

dominoFire replied:
Updated documentation.

nkolev92 commented on src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/DependencyGraphRestoreUtility.cs:
```diff
@@ -58,6 +58,19 @@ public static class DependencyGraphRestoreUtility
         /// Restore a solution and cache the dg spec to context.
         /// </summary>
+        /// <param name="solutionManager">No use</param>
+        /// <param name="dgSpec"></param>
```
Same as before.

dominoFire replied:
Fixed.

nkolev92 reviewed (commented):
This seems like a lot of tedious work :) 

A bunch of small comments.

kartheekp-ms commented:
If a method has an unused parameter and is part of shipped API, how about creating a new overloaded method without any unused parameters and marking the existing method as `Obsolete`?  I have created another issue https://github.com/NuGet/Client.Engineering/issues/876 for tracking removal of `Obsolete` members from the public API in the future versions of the product.

``` cs
// This method violates the rule because second parameter is unused in the method body.
[Obsolete("Use GetSomethingElse method that accepts one parameter")]
public static string GetSomething(int first, int second)
{
    return GetSomethingElse(first);
}

// This method satisfies the rule.
public static string GetSomethingElse(int first)
{
    return first.ToString(CultureInfo.InvariantCulture);
}
```
What do other team members think?

nkolev92 commented:
How about we make judgment calls and just remove some less likely to be used APIs? :) 
Especially if they are in something like NuGet.PackageManagement.

kartheekp-ms commented:
> How about we make judgment calls and just remove some less likely to be used APIs? :)
> Especially if they are in something like NuGet.PackageManagement.

I am fine with this suggestion too :)

zivkan commented on src/NuGet.Clients/NuGet.CommandLine/GlobalSuppressions.cs:
```diff
@@ -53,7 +51,7 @@
 [assembly: SuppressMessage("Build", "CA1062:In externally visible method 'IProjectFactory ProjectFactory.ProjectCreator(PackArgs packArgs, string path)', validate parameter 'packArgs' is non-null before using it. If appropriate, throw an ArgumentNullException when the argument is null or add a Code Contract precondition asserting non-null argument.", Justification = "<Pending>", Scope = "member", ...
 [assembly: SuppressMessage("Build", "CA1305:The behavior of 'string.Format(string, params object[])' could vary based on the current user's locale settings. Replace this call in 'ProjectFactory.WriteDetail(string, params object[])' with a call to 'string.Format(IFormatProvider, string, params object[])'.", Justification = "<Pending>", Scope = "member", Target = "~M:NuGet.CommandLine.ProjectFactor ...
-[assembly: SuppressMessage("Build", "CA1801:Parameter registryKeyPath of method GetValueFromRegistryKey is never used. Remove the parameter or use it in the method body.", Justification = "<Pending>", Scope = "member", Target = "~M:NuGet.CommandLine.RegistryKeyUtility.GetValueFromRegistryKey(System.String,System.String,Microsoft.Win32.RegistryKey,NuGet.Common.ILogger)~System.Object")]
+[assembly: SuppressMessage("Build", "CA1801:Parameter registryKeyPath of method GetValueFromRegistryKey is never used. Remove the parameter or use it in the method body.", Justification = "Shipped Public API", Scope = "member", Target = "~M:NuGet.CommandLine.RegistryKeyUtility.GetValueFromRegistryKey(System.String,System.String,Microsoft.Win32.RegistryKey,NuGet.Common.ILogger)~System.Object")]
```
I no longer remember why I made NuGet.CommandLine use the public API analyzers. Although it's packed, it goes to `tools/nuget.exe` in the package, meaning customers can't install this package in their projects and use its APIs. Hence we shouldn't have any public API concerns with this project.

dominoFire replied:
Fixed parameter usage in this case and removed unused params in NuGet.CommandLine.

dominoFire commented:
Thanks all for your comments! To address them, I will:

- Complete documentation in in public APIs when needed (estimated 27)
- Remove Public APIs from Visual Studio related packages (estimated 27)
- Add overloaded versions of Public APIs from SDK/Core libraries and deprecate those with unused parameters (estimated 3 changes)
- Regarding cancellation tokens and UI thread, I am discussing in the comment.

erdembayar commented:
> Thanks all for your comments! To address them, I will:
> 
> * Complete documentation in in public APIs when needed (estimated 27)
> * Remove Public APIs from Visual Studio related packages (estimated 27)
> * Add overloaded versions of Public APIs from SDK/Core libraries and deprecate those with unused parameters (estimated 3 changes)
> * Regarding cancellation tokens and UI thread, I am discussing in the comment.

Could you be able to change PR status to draft mode until new commits are pushed?

Commit:
Fixed some public APIs

Commit:
Fixed API in NuGet.Commands

Commit:
Addressed public APIs from NuGet.Commands

Commit:
Addressed NuGet.Build.Tasks public APIs

Commit:
Removed extra cancelation token calls

Commit:
Proofreading

dominoFire commented:
At the end, I only deprecated public APIs with unused parameters and added new public APIs with unused parameters. There are some non-Microsoft packages that depends on NuGet VS public APIs. For example https://www.nuget.org/packages/Orc.NuGetExplorer/ depends on NuGet.PackageManagement. I want to give an opportunity to developers to switch to new public APIs and remove the old ones in the next minor release.

Bugfixes:

- Fixed a telemetry bug: src/NuGet.Core/NuGet.Packaging/PackageExtraction/PackageExtractionResult.cs
- Fixed a bug in VSMsBuildProjectSystem: src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectSystems/VsMSBuildProjectSystem.cs
- Fixed a bug on RegistryKeyUtilities: src/NuGet.Clients/NuGet.CommandLine/RegistryKeyUtility.cs

Follow-up issue to remove public APIs in a later minor release: https://github.com/NuGet/Home/issues/10936

nkolev92 commented:
> I want to give an opportunity to developers to switch to new public APIs and remove the old ones in the next minor release.

We wouldn't remove the APIs in a minor release, we'd only remove them in a major release.

nkolev92 commented on src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs:
```diff
@@ -563,11 +564,29 @@ public static MsBuildToolset GetMsBuildToolset(string userVersion, IConsole cons
         /// <param name="getMsBuildPathInPathVar">delegate to provide msbuild exe discovered in path environemtnb var/s
         /// (using a delegate allows for testability)</param>
         /// <returns>directory to use for msbuild exe</returns>
+        [Obsolete("Use GetMsBuildDirectoryInternal(string, IEnumerable<MsBuildToolset>, Func<IEnvironmentVariableReader, string>) instead")]
```
nuget.exe is ILMerged and not shipped as a library. 

You can remove this method.

dominoFire replied:
Removed.

nkolev92 commented on src/NuGet.Clients/NuGet.VisualStudio.Implementation/PreinstalledRepositoryProvider.cs:
```diff
@@ -71,7 +71,24 @@ public void AddFromRegistry(string keyName, bool isPreUnzipped)
+        /// <param name="extensionId">VSPackage Extension GUID</param>
+        /// <remarks>https://docs.microsoft.com/en-us/nuget/visual-studio-extensibility/visual-studio-templates</remarks>
+        [Obsolete("Use overload without provider argument instead")]
         public void AddFromExtension(ISourceRepositoryProvider provider, string extensionId)
```
Is this an implementation of an API?

NuGet.VisualStudio.Implementation doesn't ship publicly.

dominoFire replied:
Removed.

nkolev92 commented on src/NuGet.Core/NuGet.Commands/RestoreCommand/Diagnostics/UnexpectedDependencyMessages.cs:
```diff
@@ -225,9 +224,23 @@ public static bool HasMissingLowerBound(VersionRange range)
+        /// <param name="logger">Not used</param>
+        /// <returns>A collection of log messages</returns>
+        [Obsolete("Use GetDependenciesAboveUpperBounds(List<IndexedRestoreTargetGraph>) instead")]
         public static IEnumerable<RestoreLogMessage> GetDependenciesAboveUpperBounds(List<IndexedRestoreTargetGraph> graphs, ILogger logger)
```
Given that these are restore internals and this is a major version change, i think we should *break* the API rather than add more overloads.

dominoFire replied:
Removed

nkolev92 commented on src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs:
```diff
@@ -32,11 +32,37 @@ public class LockFileBuilder
+        /// <param name="localRepositories">V3 NuGet Sources collection</param>
+        /// <param name="context">Not used</param>
+        /// <returns>A LockFile created from arguments</returns>
+        [Obsolete("Use CreateLockFile() overload without context argument instead")]
```
Same here, I want to avoid adding new API for theoretical purposes in very specific codepaths. 
We're at a major verison, I'd rather be removing APIs.

nkolev92 commented on src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreTargetGraph.cs:
```diff
@@ -85,18 +85,68 @@ public class RestoreTargetGraph : IRestoreTargetGraph
+        /// <param name="framework">Target NuGet framework to restore</param>
+        /// <returns>A <see cref="RestoreTargetGraph "/> object built from arguments</returns>
+        [Obsolete("Use Create() overload without logger argument instead")]
         public static RestoreTargetGraph Create(IEnumerable<GraphNode<RemoteResolveResult>> graphs, RemoteWalkContext context, ILogger logger, NuGetFramework framework)
```
Same feedback applies.

dominoFire replied:
Removed.

nkolev92 commented on src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/BuildAssetsUtils.cs:
```diff
@@ -421,6 +421,19 @@ public static string GetMSBuildFilePathForPackageReferenceStyleProject(PackageSp
+        /// <param name="restoreSuccess">Indicates if packages were restored successfully</param>
+        /// <param name="log">Not used</param>
+        /// <returns>A collection of props/targets files generated from restored packages</returns>
+        [Obsolete("Use GetMSBuildOutputFiles() overload without log argument instead")]
```
similar feedback applies.

nkolev92 commented on src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/NoOpRestoreUtilities.cs:
```diff
@@ -44,11 +45,23 @@ private static string GetBuildIntegratedProjectCacheFilePath(RestoreRequest requ
+        /// <param name="projectPath">Not used</param>
+        /// <returns>The combined path</returns>
+        [Obsolete("Use overload withou projectPath argument")]
         public static string GetProjectCacheFilePath(string cacheRoot, string projectPath)
```
Similar feedback as before.

I don't think removing a warning for an unused param is a good enough reason to introduce more long term work by adding more APIs.

dominoFire replied:
Removed

nkolev92 reviewed (commented):
I left much of the same feedback in these APIs. 

I think we should rather keep unused parameters than add many new public APIs that just introduce more work long term. 

We're at a major release, one of the few times we can actually clean-up old/irrelevant APIs.

Commit:
Reverting change for introduced CA2000 exception in NuGet.Protocol

ghost commented:
This PR has been automatically marked as stale because it has no activity for **7 days**. It will be closed if no further activity occurs **within another 7 days of this comment**. If it is closed, you may reopen it anytime when you're ready again, as long as you don't delete the branch.

dominoFire commented:
Reopening after addresing feedback.

dominoFire commented:
Reopened here: https://github.com/NuGet/NuGet.Client/pull/4177
